cfme/storage/object_store.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+from navmazing import NavigateToAttribute, NavigateToSibling
+
+import cfme.fixtures.pytest_selenium as sel
+from cfme.common import SummaryMixin, Taggable
+from cfme.web_ui import mixins
+from utils.appliance.implementations.ui import navigate_to, navigator, CFMENavigateStep
+from utils.appliance import Navigatable
+from cfme.web_ui import toolbar as tb
+from cfme.web_ui import Quadicon
+from utils import version
+
+
+class ObjectStore(Taggable, SummaryMixin, Navigatable):
+    """ Automate Model page of Cloud Object Stores
+
+    Args:
+        name: Name of Object Store
+    """
+
+    def __init__(self, name=None, appliance=None):
+        Navigatable.__init__(self, appliance=appliance)
+        self.name = name
+
+    def tag(self, tag, **kwargs):
+        """Tags the system by given tag"""
+        navigate_to(self, 'Details')
+        mixins.add_tag(tag, **kwargs)
+
+    def untag(self, tag):
+        """Removes the selected tag off the system"""
+        navigate_to(self, 'Details')
+        mixins.remove_tag(tag)
+
+
+@navigator.register(ObjectStore, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        if version.current_version() >= "5.7":
+            nav._nav_to_fn('Storage', 'Object Stores')(None)
+        else:
+            nav._nav_to_fn('Compute', 'Clouds', 'Object Stores')(None)
+
+
+@navigator.register(ObjectStore, 'Details')
+class Details(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        tb.select('Grid View')
+        sel.click(Quadicon.first())
```

Comment:
Details should be opening the details for a specific Object Store. You can't click on ```.first()``` it should do something like clicking on the quadicon called ```self.obj.name ```